rbd/rbd.go

```diff
@@ -1358,3 +1358,43 @@ func RemoveImage(ioctx *rados.IOContext, name string) error {
 	defer C.free(unsafe.Pointer(c_name))
 	return getError(C.rbd_remove(C.rados_ioctx_t(ioctx.Pointer()), c_name))
 }
+
+// CloneImage creates a clone of the image from the named snapshot in the
+// provided io-context with the given name and image options.
+//
+// Implements:
+//   int rbd_clone3(rados_ioctx_t p_ioctx, const char *p_name,
+//                  const char *p_snapname, rados_ioctx_t c_ioctx,
+//                  const char *c_name, rbd_image_options_t c_opts);
+func CloneImage(ioctx *rados.IOContext, parentName, snapName string,
+	destctx *rados.IOContext, name string, rio *RbdImageOptions) error {
```

Comment:
I think we should allow passing `nil` for RbdImageOptions? Defaults will be used that way.
This fails now because `rio.options` is referenced further below.

`CreateImage()` handles it not cleanly either (but better):

```golang
        if rio == nil {
                return RBDError(C.EINVAL)
        }
```

What approach do you prefer?

Reply:
Odd. I could have sworn I had that same check in this function too. I think we should be consistent so let's start off by doing the same thing in CloneImage that CreateImage does. We can then follow up with another PR that makes passing nil do something more convenient later.

Reply:
Added check like CreateImage (for now).